File: examples/cpp/example_offline.cpp
#include "../../core/cpp/notortoday.hpp"
#include <iostream>
#include <fstream>
#include <sstream>

std::string readFile(const std::string& filename) {
    std::ifstream file(filename);
    std::stringstream buffer;
    buffer << file.rdbuf();
    return buffer.str();
}

int main() {
    std::cout << "NoTorToday - C++ Example (Offline Mode)\n\n";
    
    // Create instance without auto-update
    NoTorToday checker(false);
    
    // Load sample data for testing
    try {
        std::string sampleData = readFile("../../datasets/sample-exit-addresses.txt");
        checker.parseExitAddresses(sampleData);
        
        std::cout << "Loaded " << checker.getCount() << " Tor exit nodes\n";
        std::cout << "Using sample dataset for offline testing\n\n";
        
        // Test IPs
        const char* testIPs[] = {
            "185.220.101.1",  // In sample dataset - should be Tor
            "185.220.101.2",  // In sample dataset - should be Tor
            "8.8.8.8",        // Google DNS - not Tor
            "1.1.1.1"         // Cloudflare DNS - not Tor
        };
        
        std::cout << "Testing IP addresses:\n";
        for (const auto& ip : testIPs) {
            bool isTor = checker.isTor(ip);
            std::cout << "  " << ip << ": " << (isTor ? "TOR EXIT NODE ✓" : "Not Tor ✗") << "\n";
        }
    } catch (const std::exception& e) {
        std::cerr << "Error: " << e.what() << "\n";
        return 1;
    }
    
    return 0;
}

File: examples/dart/example_offline.dart
import 'dart:io';
import '../../../core/dart/notortoday.dart';

void main() async {
  print('NoTorToday - Dart Example (Offline Mode)\n');
  
  // Create instance without auto-update
  final checker = NoTorToday(autoUpdate: false);
  
  // Load sample data for testing
  try {
    final sampleData = await File('../../datasets/sample-exit-addresses.txt').readAsString();
    checker.parseExitAddresses(sampleData);
    
    print('Loaded ${checker.getCount()} Tor exit nodes');
    print('Using sample dataset for offline testing\n');
    
    // Test IPs
    final testIPs = [
      '185.220.101.1',  // In sample dataset - should be Tor
      '185.220.101.2',  // In sample dataset - should be Tor
      '8.8.8.8',        // Google DNS - not Tor
      '1.1.1.1',        // Cloudflare DNS - not Tor
    ];
    
    print('Testing IP addresses:');
    for (final ip in testIPs) {
      final isTor = checker.isTor(ip);
      final status = isTor ? 'TOR EXIT NODE ✓' : 'Not Tor ✗';
      print('  $ip: $status');
    }
  } catch (e) {
    stderr.writeln('Error: $e');
  }
}
